include/internal/params.h

```diff
@@ -36,3 +40,29 @@ int ossl_param_get1_octet_string(const OSSL_PARAM *params, const char *name,
 int ossl_param_get1_concat_octet_string(const OSSL_PARAM *params, const char *name,
                                         unsigned char **out, size_t *out_len,
                                         size_t maxsize);
+
+/*
+ * Merge multiple parameter lists together.
+ * Duplicates are removed but not in a specified order.
+ */
+OSSL_PARAM *ossl_param_merge(size_t n, const OSSL_PARAM **plists);
+
+/*
+ * Duplicate a parameter list removing some items.
+ * The `omit' list is rearranged but left otherwise unchanged.
+ */
+OSSL_PARAM *ossl_param_dup_omit(const OSSL_PARAM *src, OSSL_PARAM *omit);
+
+/*
+ * Count the number of elements in a parameter list.
+ */
+size_t ossl_param_nelem(const OSSL_PARAM *plist);
+
+/*
+ * Check that a param list is NULL or empty.
+ */
+static ossl_inline int ossl_param_is_empty(const OSSL_PARAM params[])
```

Comment:
There is another one of these declared in the provider includes.

Reply:
There isn't anymore.  I moved this to a more visible header.